client/components/Facilitator/Components/Search/index.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+import { Dropdown, Form, Grid, Input, List } from 'semantic-ui-react';
+// TODO
+// import PropTypes from 'prop-types';
+import Search from '@client/components/Search';
+
+const SearchResults = ({ view, results }) => {
+    if (!results.length) {
+        return null;
+    }
+    return results.map(record => {
+        const {
+            description,
+            email,
+            ended_at,
+            id,
+            title = '',
+            username = ''
+        } = record;
+        return view === 'users' ? (
+            <List.Item fluid="true" key={id}>
+                <List.Header as="h3">{username}</List.Header>
+                <List.Content>
+                {email}<br />
+                Clicking (or not?) on this item should display controls
+                for:
+                    <ul>
+                        <li>viewing this participant's runs.</li>
+                        <li>initiating a session</li>
+                    </ul>
+                </List.Content>
+            </List.Item>
+        ) : (
+            <List.Item fluid="true" key={id}>
+                <List.Header as="h3">
+                    {username} participated in &quot;{title}&quot; on {ended_at}
+                </List.Header>
+                <List.Content>{description}</List.Content>
+            </List.Item>
+        );
+    });
+};
+
+const availableViews = [
+    {
+        key: 'users',
+        text: 'Users',
+        value: 'users'
+    },
+    {
+        key: 'runs',
+        text: 'Runs',
+        value: 'runs'
+    }
+];
+
+const defaultAvailableViewValue = availableViews[0].value;
+
+class FacilitatorSearch extends Search {
+    constructor(props) {
+        super(props);
+
+        this.timeout = null;
+        this.state = {
+            view: defaultAvailableViewValue,
+            keyword: '',
+            results: []
+        };
+
+        this.onChange = this.onChange.bind(this);
+        this.onDropdownChange = this.onDropdownChange.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onDropdownChange(event, data = {}) {
+        this.setState({
+            view: data.value
+        });
+    }
+
+    onChange(event, data = {}) {
+        this.setState({
+            keyword: data.value
+        });
+
+        this.onSubmit();
+    }
+
+    async onSubmit() {
+        const { keyword, view } = this.state;
+        if (keyword && keyword.length > 1 && view) {
+            const response = await Search.query(view, { keyword });
+            const { error, results } = response;
+            if (error) {
+                // TODO: implement error message display?
+                // const { message, stack } = response;
```

Comment:
Perhaps this would be a good place to use the same pattern as the editor and add a message component to the tabs. Another argument for uplifting the tabs pattern into its own component, as you mentioned offline @rwaldron.

Reply:
Let's do that in a follow up!